public/app/features/api-keys/ApiKeysPage.tsx

```diff
@@ -142,8 +143,14 @@ export class ApiKeysPageUnconnected extends PureComponent<Props, State> {
   };
 
   onHideApiKeys = async () => {
-    await this.props.hideApiKeys();
-    window.location.reload();
+    try {
+      await this.props.hideApiKeys();
+      let serviceAccountsUrl = '/org/serviceaccounts';
+      locationService.push(serviceAccountsUrl);
+      window.location.reload();
```

Comment:
Had to add the reload as otherwise the tab remains after changing the location, this will make sure that user does not need to reload the page manually.